pkg/karmadactl/addons/install/install_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package install
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl/addons/descheduler"
+	"github.com/karmada-io/karmada/pkg/karmadactl/addons/estimator"
+	addonsinit "github.com/karmada-io/karmada/pkg/karmadactl/addons/init"
+	"github.com/karmada-io/karmada/pkg/karmadactl/addons/metricsadapter"
+	"github.com/karmada-io/karmada/pkg/karmadactl/addons/search"
+)
+
+func TestInstall(t *testing.T) {
+	// Call the function to install addons.
+	Install()
+
+	tests := []struct {
+		name          string
+		key           string
+		expectedAddon interface{}
+	}{
+		{
+			name:          "Install_WithKarmadaDeschedulerAddon_Installed",
+			key:           "karmada-descheduler",
```

Comment:
It is recommended that you change the component name to use constants so that the entire code base can be unified. If you think it is feasible, you can mention a separate pr.

Reply:
Good idea! I've mentioned this in a separate PR #5908. I would love to receive any feedback! 🙏